activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,34 @@
+*   Add `primary-key-default` option to create table migrations.
+
+    This allows setting the `default` argument when creating a migration
+
+    ```bash
+    rails generate migration create_books --primary-key-type=uuid --primary-key-default="uuid_generate_v4()"
+    ```
+
+    ```ruby
+    class CreateBooks < ActiveRecord::Migration[7.0]
+      def change
+        create_table :books, id: :uuid, default: "uuid_generate_v4()" do |t|
```

Comment:
`default: ` syntax in create_table method is it an existing option? or just introduced in this PR?
when I checked in https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-create_table
there is no `default` setting for create_table method
if it's just introduced in this PR, I would prefer to use clearer name, maybe `default_uuid`
what do you think?

Reply:
These are `options` passed through for `table_definition.set_primary_key` and uses this allowlist: https://github.com/rails/rails/blob/05cb63abdaf6101e6c8fb43119e2c0d08e543c28/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L1508-L1510 
- which ultimately is just a column definition.